### PR TITLE
Add remaining ASR stmt nodes docs

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Ref https://github.com/lfortran/lfortran/issues/842

### CODE REVIEW

File: doc/src/asr/asr_nodes/statement_nodes/flush.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+`label` contains label of the branch target statement that receives control if
+an error occurs.
+
+`unit` contains external unit specifier.
+
+`err` contains error message.
+
+`iomsg` contains an explanatory message if an I/O error occurs.
+
+`iostat` contains
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/flush.md:25]: @certik: what would this contain?

File: doc/src/asr/asr_nodes/statement_nodes/listappend.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# ListAppend
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/listappend.md:1]: @certik: is `ListAppend` implemented? 

https://github.com/lfortran/lfortran/search?q=ListAppend&type=code

File: doc/src/asr/asr_nodes/statement_nodes/setinsert.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# SetInsert
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/setinsert.md:1]: @certik: is this implemented?

https://github.com/lfortran/lfortran/search?q=SetInsert&type=code

File: doc/src/asr/asr_nodes/statement_nodes/listremove.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+```fortran
+```
+
+ASR:
+
+```fortran
+```
+
+## See Also
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/listremove.md:34]: @certik: are the following implemented?

- `listappend`
- `listclear`
- `listinsert`
- `setinsert`
- `setremove`
- `dictinsert`

https://github.com/lfortran/lfortran/search?q=searchterm&type=code

Comment (certik) [reply doc/src/asr/asr_nodes/statement_nodes/listremove.md:34]: Not in LFortran, so you can skip those. They are currently only used in LPython.

File: doc/src/asr/asr_nodes/statement_nodes/listappend.md (hunk 2 above)
Comment (certik) [reply doc/src/asr/asr_nodes/statement_nodes/listappend.md:1]: Not in LFortran.

File: doc/src/asr/asr_nodes/statement_nodes/flush.md (hunk 1 above)
Comment (certik) [reply doc/src/asr/asr_nodes/statement_nodes/flush.md:25]: I think this is the return value of the operation. Status code.

**Review by certik — APPROVED** [review]
Looks great!

File: doc/src/asr/asr_nodes/statement_nodes/setinsert.md (hunk 3 above)
Comment (certik) [reply doc/src/asr/asr_nodes/statement_nodes/setinsert.md:1]: Not in LFortran.

**Review by certik — APPROVED** [review]
I think this looks good. @tapaswenipathak do you want to change anything else, or can we merge?

**Comment by tapaswenipathak** [comment]
On Sun, 4 Dec 2022, 13:18 Ondřej Čertík, ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> I think this looks good. @tapaswenipathak
> <https://github.com/tapaswenipathak> do you want to change anything else,
> or can we merge?
>

yes!

**Pull request merged**